packages/radix-ui-themes/src/helpers/responsive.ts

```diff
@@ -0,0 +1,181 @@
+import { hasOwnProperty } from './has-own-property';
+import type { StringOrValue } from './string-or-value';
+
+const breakpoints = ['initial', 'xs', 'sm', 'md', 'lg', 'xl'] as const;
+type Breakpoints = (typeof breakpoints)[number];
+type Responsive<T> = T | Partial<Record<Breakpoints, T>>;
+
+interface GetResponsiveStylesOptions {
+  className: string;
+  customProperty: `--${string}`;
+  value: Responsive<StringOrValue<string>> | Responsive<string> | undefined;
+  propValues: string[] | readonly string[];
+  parseValue?: (value: string) => string | undefined;
+}
+
+function getResponsiveStyles({ className, customProperty, ...args }: GetResponsiveStylesOptions) {
+  const classNames = getResponsiveClassNames({
+    allowArbitraryValues: true,
+    className,
+    ...args,
+  });
+
+  const customProperties = getResponsiveCustomProperties({ customProperty, ...args });
+  return [classNames, customProperties] as const;
+}
+
+interface GetResponsiveClassNamesOptions {
+  allowArbitraryValues?: boolean;
+  className: string;
+  value: Responsive<StringOrValue<string>> | Responsive<string> | undefined;
+  propValues: string[] | readonly string[];
+  parseValue?: (value: string) => string | undefined;
+}
+
+function getResponsiveClassNames({
+  allowArbitraryValues,
+  value,
+  className,
+  propValues,
+  parseValue = (value) => value,
+}: GetResponsiveClassNamesOptions): string | undefined {
+  const classNames: string[] = [];
+
+  if (!value) {
+    return '';
```

Comment:
nit: That would feel more accurate to the return signature.
```suggestion
    return undefined;
```